VDA-StorageNode-Registry/test/index.test.ts

```diff
@@ -0,0 +1,939 @@
+                        datacenterIds[0],
+                        0,
+                        invalidLongValues[i]);
+                    await expect(
+                        contract.addNode(nodeInfo, "0x00", "0x00", "0x00")
+                    ).to.be.revertedWithCustomError(contract, "InvalidLongitude")
+                }
+            })
+
+            it("Failed: No trusted signer",async () => {
+                const nonce = await contract.nonce(user.address);
+                
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, trustedSigner);
+
+                await expect(
+                    contract.addNode(storageNode, requestSignature, requestProof, authSignature)
+                ).to.be.revertedWithCustomError(contract, "NoSigners");
+            })
+
+            it("Failed: Invalid auth signature",async () => {
+                await contract.addTrustedSigner(trustedSigner.address);
+
+                const badSigner = Wallet.createRandom();
+
+                const nonce = await contract.nonce(user.address);
+
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, badSigner);
+
+                await expect(
+                    contract.addNode(storageNode, requestSignature, requestProof, authSignature)
+                ).to.be.revertedWithCustomError(contract, "InvalidSignature");
+            })
+
+            it("Success", async () => {
+                const nonce = await contract.nonce(user.address);
+
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, trustedSigner);
+
+                const tx = await contract.addNode(storageNode, requestSignature, requestProof, authSignature);
+
+                await expect(tx).to.emit(contract, "AddNode").withArgs(
+                    storageNode.didAddress,
+                    storageNode.endpointUri,
+                    storageNode.countryCode,
+                    storageNode.regionCode,
+                    storageNode.datacenterId,
+                    storageNode.lat,
+                    storageNode.long
+                );
+            })
+
+            it("Failed: Registered didAddress & endpointURI", async () => {
+                // Registered DID
+                await checkAddNode(storageNode, user, trustedSigner, false, "InvalidDIDAddress");
+                
+                // Registered EndpointURI
+                {
+                    const anotherUser = Wallet.createRandom();
+
+                    const nodeInfo = {...storageNode};
+                    nodeInfo.didAddress = anotherUser.address;
+
+                    await checkAddNode(nodeInfo, anotherUser, trustedSigner, false, "InvalidEndpointUri");
+                }
+
+            })
+
+            it("Failed: Removing didAddress & endpointURI", async () => {
```

Comment:
Please rename this test or add more detail, I'm unclear what is eing tested.

Reply:
DidAddress & endpoints that are in pending removal states can't be used in `addNode()` function.
Renamed to "Failed: didAddress & endpointURI in pending `removal` status".